projects/packages/connection/src/webhooks/class-authorize-redirect.php

```diff
@@ -97,44 +98,57 @@ function ( $domains ) {
 	}
 
 	/**
-	 * Create the Jetpack authorization URL. Copied from Jetpack class.
+	 * Create the Jetpack authorization URL.
+	 *
+	 * @since $$next-version$$ Added optional $from and $raw parameters.
 	 *
 	 * @param bool|string $redirect URL to redirect to.
+	 * @param bool|string $from     If not false, adds 'from=$from' param to the connect URL.
+	 * @param bool        $raw If true, URL will not be escaped.
 	 *
 	 * @todo Update default value for redirect since the called function expects a string.
 	 *
 	 * @return mixed|void
 	 */
-	public function build_authorize_url( $redirect = false ) {
+	public function build_authorize_url( $redirect = false, $from = false, $raw = false ) {
```

Comment:
At this point I'm wondering if `build_authorize_url` shouldn't be merged into `get_authorization_url`, but I'm not familiar enough with this to be confident that change can be done.